frontend/src/views/admin/assignments/offer-actions.js

```diff
@@ -136,37 +191,85 @@ function OfferActionButtons(props) {
             </ActionButton>
             <ActionButton
                 icon={<FaEnvelope />}
-                onClick={emailOffers}
+                onClick={confirmOfferEmail}
                 disabled={!actionPermitted.canEmail}
             >
                 Email Offer
             </ActionButton>
             <ActionButton
                 icon={<FaUserClock />}
-                onClick={nagOffers}
+                onClick={confirmOfferNag}
                 disabled={!actionPermitted.canNag}
             >
                 Nag Offer
             </ActionButton>
             <ActionButton
                 icon={<FaCheck />}
-                onClick={acceptOffers}
+                onClick={confirmOfferAccept}
                 disabled={!actionPermitted.canAccept}
             >
                 Set as Accepted
             </ActionButton>
             <ActionButton
                 icon={<FaBan />}
-                onClick={rejectOffers}
+                onClick={confirmOfferReject}
                 disabled={!actionPermitted.canReject}
             >
                 Set as Rejected
             </ActionButton>
-            <MultiWithdrawOfferConfirmation
+            <MultiManipulateOfferConfirmation
+                data={selectedAssignments}
+                visible={showCreateConfirmation}
+                setVisible={setShowCreateConfirmation}
+                manipulateOffers={createOffers}
+                titleMsg="Creating Multiple Offers"
+                alertMsg={`You are creating the following ${selectedAssignments.length} offers:`}
+                confirmBtnMsg={`Create ${selectedAssignments.length} Offers`}
+            />
```

Comment:
We're getting a lot of code in this file...can we split the buttons into their own file and have a `<ConfirmWithDialogActionButton />` which is a component returning `<React.Fragment><ActionButton .../><ConfirmationDialog ... /></React.Fragment>`?